uint32 AgentID
CellPosition Start
CellPosition Goal